### ISSUE TITLE
Implement Terraform

### ISSUE THREAD

**Issue by Karvel** [Shift3/boilerplate-client-angular#6]
## Desired Use Case

The boilerplate should have a working Terraform config to set up AWS and S3 for sandbox deployments. Ideally, someone should be able to have a codebase that they can deploy to AWS in two steps (1. Set up AWS through Terraform 2. Deploy the sandbox to AWS).

## Actual/Current Behavior

The feature is not yet implemented.

## Steps to Implement This Feature

  1. Add a working Terraform configuration.
  2. Add a deployment script to `package.json`.
  3. Update the documentation with deployment instructions.

## Learning

Basically just the brain of @mwallert.
